wordpress-mobile/WordPress-Android
# Issue/10639 material site settings

Fixes #10639

This PR moves Site Settings to Material Theme and adds Dark Theme support.

Site Settings look pretty simple, but they are super-old, and there are a lot of scattered changes. Luckily it's pretty easy to confirm that everything looks ok.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on the "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature, that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

![comparison_settings](https://user-images.githubusercontent.com/728822/72197774-051e9980-33da-11ea-9a2e-969cdb18a017.png)

To test:
- Ideally, navigate to Site Settings and confirm the visual and functional integrity of each element in both Light and Dark themes.
The theme can be changed from App Settings.

PR submission checklist:
